src/services/data-exporter/json.ts

```diff
@@ -21,82 +21,43 @@ import Realm from 'realm';
 
 import { IExportEngine } from '.';
 
-function serializeObject(object: { [key: string]: any } & Realm.Object) {
-  // This is an object reference
-  const objectSchema = object.objectSchema();
-  if (objectSchema.primaryKey) {
-    return object[objectSchema.primaryKey];
-  } else {
-    // Shallow copy the object
-    return RealmObjectToJSON.call(object);
-  }
-}
+const INDENTATION_SPACES = 2;
 
-function serializeValue(propertyName: string, value: any) {
-  if (
-    value === null ||
-    typeof value === 'string' ||
-    typeof value === 'number' ||
-    typeof value === 'boolean'
-  ) {
-    return value;
-  } else if (value instanceof Date) {
-    return value.toISOString();
-  } else if (value instanceof ArrayBuffer) {
-    return Buffer.from(value).toString('base64');
-  } else if (
-    typeof value === 'object' &&
-    typeof value.objectSchema === 'function'
-  ) {
-    return serializeObject(value);
-  } else if (typeof value === 'object' && typeof value.length === 'number') {
-    if (value.type === 'object') {
-      // A list of objects
-      return value.map((item: any) => {
-        if (typeof item === 'object') {
-          return serializeObject(item);
-        } else {
-          return item;
-        }
-      });
-    } else {
-      // A list of primitives
-      return [...value];
-    }
-  } else {
-    throw new Error(
-      `Failed to serialize '${propertyName}' field of type ${typeof value}`,
-    );
-  }
-}
+type ResultMap = {
+  [key: string]: Realm.Results<Realm.Object>;
+};
 
-function RealmObjectToJSON(this: { [key: string]: any } & Realm.Object) {
-  const values: { [key: string]: any } = {};
-  for (const propertyName of Object.getOwnPropertyNames(this)) {
-    const value = this[propertyName];
-    if (propertyName === '_realm' || typeof value === 'function') {
-      continue; // Skip this property
-    } else {
-      values[propertyName] = serializeValue(propertyName, value);
+const serialize = (map: ResultMap) => {
+  try {
+    return JSON.stringify(map, null, INDENTATION_SPACES);
+  } catch (err) {
+    if (
+      err instanceof TypeError &&
+      err.message.startsWith('Converting circular structure to JSON')
+    ) {
+      return JSON.stringify(
+        map,
+        Realm.JsonSerializationReplacer,
```

Comment:
Any reason that we're not just always calling with the `JsonSerializationReplacer`? If yes, I think that would be a good candidate for a comment.

Reply:
It will add `$refId` to entities that does not have a `primaryKey` for reference. I'll add a comment.

Reply:
Added.